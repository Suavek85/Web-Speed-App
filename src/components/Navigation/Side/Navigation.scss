@import '../../../scss/styles';

$block: nav-wrapper;
$blockBackground: nav-wrapper-background;
$blockCloseWrapper: nav-wrapper-close;

.#{$block} {
  position: absolute;
  top: 0;
  left: -300px;
  z-index: 100;
  display: block;
  width: 0;
  height: 100vh;
  background-color: $color-white;
}

.#{$blockBackground} {
  position: fixed;
  top: 0;
  right: 0;
  left: 0;
  z-index: 10;
  display: none;
  width: 100%;
  height: 100%;
  background-color: $color-black;
  opacity: 0;
}

.#{$blockCloseWrapper} {
  margin-left: 15px;
}

:export {
  block: $block;
  blockBackground: $blockBackground;
  blockCloseWrapper: $blockCloseWrapper;
}
