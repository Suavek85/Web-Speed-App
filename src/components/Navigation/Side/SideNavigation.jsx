import React from 'react'
import { useSpring, animated } from 'react-spring'
import { useSelector, useDispatch } from "react-redux"
import { useMediaQuery } from 'react-responsive'
import { desktopWidth } from'../../../scss/mediaqueries'
import Button from '../../../utils/Button/Button'
import ToggleSwitch from '../../ToggleSwitch/ToggleSwitch'
import MenuLink from '../../../utils/MenuLink/MenuLink'
import { SIGN_IN } from '../../../constants/content'
import styles from './SideNavigation.scss'
import { 
  collapseNavConfig, 
  expandNavConfig, 
  collapseBackgroundConfig, 
  expandBackgroundConfig } from './animatedStyles'
import { navigationStates } from '../../../constants/navigationStates'


function SideNavigation() {

  const getToggleState = useSelector(state => state.toggleReducer.toggleNavigation)

  //using useSpring instead of useTransition becouse of react-spring beta version bug
  const collapseNav = useSpring(collapseNavConfig)
  const expandNav = useSpring(expandNavConfig)
  const collapseBackground = useSpring(collapseBackgroundConfig)
  const expandBackground = useSpring(expandBackgroundConfig)


  const dispatch = useDispatch()
  const handleCollapseMenu = () => dispatch({ type: navigationStates.COLLAPSE })

  const getAnimation = (getToggleState, expandAnimation, collapseAnimation) => {
    if (getToggleState === navigationStates.EXPAND) return expandAnimation
    if (getToggleState === navigationStates.COLLAPSE ) return collapseAnimation
    return
  }

  const isDesktop = useMediaQuery(desktopWidth)

  return (
    <>
      <animated.div 
        style={ getAnimation(getToggleState, expandBackground, collapseBackground) } 
        className={ styles.blockBackground } >
      </animated.div>
      <animated.div 
        style={ getAnimation(getToggleState, expandNav, collapseNav) } 
        className={ styles.block } 
      >
        <div className={ styles.blockContent } style={{transform: 'translate(15px, 15px)',}}>
          <Button 
            btnStyle='oval' 
            handleOnClick={ handleCollapseMenu }
            isDesktopCloseContent
          />
          {!isDesktop && (
            <>
              <MenuLink linkPath="/signin" content={ SIGN_IN } />
              <ToggleSwitch />
            </>
          )}
        </div>
      </animated.div>
    </>
  )}
  
export default React.memo(SideNavigation)
