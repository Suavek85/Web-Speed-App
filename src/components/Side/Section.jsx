import React from 'react'
import classNames from 'classnames/bind'

import LightBulbUI from '../svgs/LightBulb/LightBulbUI'
import styles from './Section.scss'

export default function Section(props) {

  const { content, data, position } = props
  const { header, mainContent, result } = content

  //STYLES
  let cx = classNames.bind(styles)
  let getClass = cx({
    blockWrapper: true,
    blockGeneric: position === 'generic',
    blockTop: position === 'top',
    blockBottom: position === 'bottom',
  })

  return (
    <div className={ getClass } >
      <div className={ styles.contentWrapper }>
        <h2>
          { header }
          <LightBulbUI />
        </h2>
        <p>
          { mainContent }
        </p>
        <p>
          { result }
          <span>
            { data }
          </span>
        </p>
      </div>
    </div>
  )
}
