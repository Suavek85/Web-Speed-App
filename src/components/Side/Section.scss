@import '../../scss/styles.scss';

.section__wrapper {
    width: 100%;
    height: 100%;
    display: flex;
    flex-direction: column;
    background-color: $color-ivory;
    &--generic {
      border-bottom: solid 1px #E4F0F7;
    }
    &--top {
      border-bottom: solid 1px #E4F0F7;
    }
    &--bottom {
      border-bottom: solid 1px #E4F0F7;
    }
}

.section__content-wrapper {
    display: flex;
    position: relative;
    flex-direction: column;
    padding: 0px 30px;
    > h2 {
      color: $color-main-font;
      font-size: 19px;
      margin-bottom: 0;
    }
    p {
      font-size: 13px;
      color: grey;
      display: flex;
      flex-direction: row;
      align-items: center;
    }
    p span {
      font-weight: 500;
      padding-left: 5px;
    }

    &--alert {
      color: $color-alert;
    }

    &--warning {
      color: $color-warning-loading;
    }

    &--info {
      color: $color-green-success;
      
    }
}

.section__lightbulb {
  position: absolute;
  right: 5px; 
  top: 5px;
}

.score__para {
  display: flex; 
  align-items: center;
}

:export {
    blockWrapper: section__wrapper;
    blockGeneric: section__wrapper--generic;
    blockTop: section__wrapper--top;
    blockBottom: section__wrapper--bottom;
    blockLightbulb: section__lightbulb;
    contentWrapper: section__content-wrapper;
    scorePara: score__para;
    blockAlert: section__content-wrapper--alert;
    blockWarning: section__content-wrapper--warning;
    blockInfo: section__content-wrapper--info;
}
