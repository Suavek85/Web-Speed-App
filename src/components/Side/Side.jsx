import React from 'react'
import classNames from 'classnames/bind'
import { useMediaQuery } from 'react-responsive'

import Main from './Main/Main'
import Middle from './Middle/Middle'
import Top from './Top/Top'
import Bottom from './Bottom/Bottom'
import { tabletWidth, mobileWidth } from'../../scss/mediaqueries'
import styles from './Side.scss'

export default function Side() {

  //STYLES
  const cx = classNames.bind(styles)
  const getClass = cx({
    block: true,
    blockTablet: useMediaQuery(tabletWidth),
    blockMobile: useMediaQuery(mobileWidth),
  })

  return (
    <div className={ getClass } >
      <Main />
      <Top />
      <Middle />
      <Bottom />
    </div>
  )
}