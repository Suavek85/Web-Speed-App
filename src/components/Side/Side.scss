//TODO: rename class
//TODO: at 1000px width 100% and flex-direction row, remove width and height, margin everywhere 20px

.results__wrapper { 
    display: flex;
    width: 30%;
    height: 100vh;
    flex-direction: column;
}

:export {
    block: results__wrapper;
}