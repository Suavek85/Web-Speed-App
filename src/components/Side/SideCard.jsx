import React from 'react'
import ReactTooltip from "react-tooltip"
import classNames from 'classnames/bind'
import { useMediaQuery } from 'react-responsive'
import { useSelector } from "react-redux"
import { appStates } from '../../constants/states'
import { desktopWidth } from'../../scss/mediaqueries'
import SkeletonCircle from '../Skeleton/SkeletonCircle'
import InfoCircle from '../svgs/InfoCircle/InfoCircle'
import SkeletonText from '../Skeleton/SkeletonText'
import LightBulbUI from '../svgs/LightBulb/LightBulbUI'
import ScoreCircle from '../svgs/ScoreCircle/ScoreCircle'
import styles from './SideCard.scss'


export default function Section(props) {

  const { content, data, position } = props
  const { header, mainContent, result } = content 

  const getAppState = useSelector(state => state.stateReducer.getAppState)
  const loadingStatus = getAppState === appStates.LOADING
  const isDesktop = useMediaQuery(desktopWidth)

  //STYLES
  const cx = classNames.bind(styles)
  const getClass = cx({
    block: true,
    blockGeneric: position === 'generic',
    blockTop: position === 'top',
    blockBottom: position === 'bottom',
  })

  const getColorClass = arg => cx({
    blockContentAlert: arg === 'SLOW',
    blockContentInfo: arg === 'FAST',
    blockContentWarning: arg === 'AVERAGE',
  })

  const getDescrptionWrapperClass = cx({
    blockDesc: true,
    blockGenericDesc: position === 'generic',
    blockResultDesc: position !== 'generic',
  })
  

  // GENERIC ELEMENTS

  const genericDescription = (
    <p>{ mainContent }</p>
  )

  const genericPara = (el, index) => (
    <p key={index}>{ result[index]  } 
      <span style={{ marginLeft:'5px' }} className={ getColorClass(el)}>
        { el !== '?' && el }
        { el === '?' && loadingStatus && <SkeletonText />}
      </span>
    </p>
  )

  //RESULTS ELEMENTS

  const description = (
    <>
      <span style={{marginLeft: '7px'}}>
        <InfoCircle mainContent={mainContent} />
      </span>
      <ReactTooltip id={ mainContent } type='info'>
        <span>{ mainContent }</span>
      </ReactTooltip>
    </>
  )

  const scorePara = (el, index) => { 
    return el !== '?' && !loadingStatus && (
      <p key={index}>
        Score: 
        <span>
          <ScoreCircle score={el} />
        </span>
      </p>
    )
  }

  const scoreParaLoading = (el, index) => { 
    return el === '?' && loadingStatus && (
      <p className={scorePara} key={index}>
        Score: 
        <span>
          <SkeletonCircle />
        </span>
      </p>
    )
  }

  return (
    <div className={ getClass } >
      { isDesktop && position === 'generic' && <LightBulbUI /> }
      <div className={ styles.blockContent }>
        <div className={ getDescrptionWrapperClass }>
          <h2>
            { header }
          </h2>
          { position === 'generic' && genericDescription }
          { position !== 'generic' && description}
        </div>
        { position === 'generic' && data.map(genericPara) }
        { position !== 'generic' && data.map(scorePara) }
        { position !== 'generic' && data.map(scoreParaLoading) }
      </div>
    </div>
  )
}
