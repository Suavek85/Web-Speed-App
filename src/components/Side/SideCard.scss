@import '../../scss/styles';

$block: sidecard;
$blockGeneric: modifier($block, generic);
$blockTop: modifier($block, top);
$blockBottom: modifier($block, bottom);
$blockContent: elem($block, content);
$blockContentAlert: modifier($blockContent, alert);
$blockContentWarning: modifier($blockContent, warning);
$blockContentInfo: modifier($blockContent, info);
$blockDesc: elem($block, desc);
$blockGenericDesc: modifier($blockDesc, generic);
$blockResultDesc: modifier($blockDesc, result);
$blockBulb: elem($block, bulb);
$blockScore: elem($block, score);

.#{$block} {
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  background-color: var(--color-inner-background);

  &--generic {
    border-bottom: 1px solid var(--color-main-background);
  }

  &--top {
    border-bottom: 1px solid var(--color-main-background);
  }

  &--bottom {
    border-bottom: 1px solid var(--color-main-background);
  }
}

.#{$blockContent} {
  position: relative;
  display: flex;
  flex-direction: column;
  padding: 0 30px;

  > div > h2 {
    margin-bottom: 0;
    font-size: em(17);
    color: var(--color-header-font);
  }

  p {
    display: flex;
    flex-direction: row;
    align-items: center;
    margin-top: 10px;
    font-size: em(13);
    color: var(--color-very-small-font);
  }

  p span {
    font-weight: 500;
  }

  &--alert {
    color: $color-alert;
  }

  &--warning {
    color: $color-warning-loading;
  }

  &--info {
    color: $color-green-success;
  }
}

.#{$blockDesc} {
  display: flex;
  margin-bottom: 5px;

  &--generic {
    align-items: flex-end;
  }

  &--result {
    align-items: flex-end;
  }
}

.#{$blockBulb} {
  position: absolute;
  top: 5px;
  right: 5px;
}

:export {
  block: $block;
  blockGeneric: $blockGeneric;
  blockTop: $blockTop;
  blockBottom: $blockBottom;
  blockContent: $blockContent;
  blockContentAlert: $blockContentAlert;
  blockContentWarning: $blockContentWarning;
  blockContentInfo: $blockContentInfo;
  blockDesc: $blockDesc;
  blockGenericDesc: $blockGenericDesc;
  blockResultDesc: $blockResultDesc;
  blockBulb: $blockBulb;
}
