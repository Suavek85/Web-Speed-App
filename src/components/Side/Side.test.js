import React from 'react'
import { shallow } from 'enzyme'
import Side from './Side'

describe('Side snapshot', () => {

  it('should match the snapshot', () => {
    const wrapper = shallow(<Side />)
    expect(wrapper).toMatchSnapshot()
  })

})