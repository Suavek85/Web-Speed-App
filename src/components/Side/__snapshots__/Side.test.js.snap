// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Side snapshot should match the snapshot 1`] = `
<aside
  className="block"
>
  <Main />
  <Top />
  <Middle />
  <Bottom />
</aside>
`;
